module-core/module-client/src/main/kotlin/almumol/ossori/core/client/github/GithubClient.kt

```diff
@@ -53,6 +55,43 @@ class GithubClient(
         return getFromGithub(uri)
     }
 
+    //PAT 를 사용해야 정상적으로 동작하고, 사용해도 분 당 10회라는 낮은 사용량이 제공됨
+    fun getContentLocation(repositoryOwner: String, repositoryName: String, content: String): GithubSearchResponse {
+        val filterQuery = "?q=repo:$repositoryOwner/$repositoryName+filename:$content"
+        val uri = "${githubClientProperties.searchBaseUrl}/code$filterQuery"
+        return getFromGithub(uri)
+    }
+
+    fun getContent(repositoryOwner: String, repositoryName: String, contentLocation: String): GithubContentResponse {
+        val uri =
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/$contentLocation"
+        return getFromGithub(uri)
+    }
+
+    fun getContributing(repositoryOwner: String, repositoryName: String): GithubContentResponse {
+        val uris = listOf(
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/CONTRIBUTING.md",
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/.github/CONTRIBUTING.md",
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/docs/CONTRIBUTING.md",
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/contributing.md",
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/.github/contributing.md",
+            "${githubClientProperties.repositoryBaseUrl}/$repositoryOwner/$repositoryName/contents/docs/contributing.md"
+        )
+
+        for (uri in uris) {
+            try {
+                return getFromGithub(uri)
+            } catch (e: GithubResponseException) {
+                if (e.statusCode != HttpStatus.NOT_FOUND) {
+                    throw e
+                }
+            }
+        }
+
+        throw GithubResponseException(HttpStatus.NOT_FOUND, "CONTRIBUTING.md not found in repository $repositoryOwner/$repositoryName")
+    }
+
```

Comment:
firstNotNullOfOrNull라는 놈이 있네여 
for 루프 없앨 수 있나봐여
```suggestion
return uris.firstNotNullOfOrNull { uri ->
    try {
        getFromGithub(uri)
    } catch (e: GithubResponseException) {
        if (e.statusCode != HttpStatus.NOT_FOUND) throw e
        null
    }
} ?: throw GithubResponseException(
    HttpStatus.NOT_FOUND,
    "CONTRIBUTING.md not found in repository $repositoryOwner/$repositoryName"
)
```

Reply:
굳이 for를 쓸 필요가 없었네요 ㅎㅎ